### PR TITLE
Fix to resolve node loading failure after unexpected node termination

### PR DESCRIPTION
**Pull request by aidan-kwon**
### Root cause 
In the `loadLastState` method, `bc.currentHeader` can be updated after storing `bc.currentBlock`. This mismatch causes an expected result during the node initializing process. 

https://github.com/klaytn/klaytn/blob/5d7a03b7bb2d834393d26719e52896ed3eeb1a06/blockchain/blockchain.go#L468-L478

`Governance.UpdateParams` requires the head block number of the canonical chain, but it had used the head of header chain that is used for light client usually. This PR does
 1. Replace the use of `bc.currentHeader` to `bc.currentBlock`
 2. Add a parameter indicating a block number for `UpdateParams` method

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: governance/interface.go
Comment (blukat29) [inline governance/interface.go:50]:
```suggestion
	// guarantee that Params() works correctly.
```

File: governance/contract_test.go
Comment (blukat29) [inline governance/contract_test.go:261]:
```suggestion
		err := e.UpdateParams(sim.BlockChain().CurrentBlock().NumberU64())
```

File: governance/interface.go
Comment (aidan-kwon) [reply governance/interface.go]: applied. Thanks!

File: governance/contract_test.go
Comment (aidan-kwon) [reply governance/contract_test.go]: applied. Thanks!

File: governance/interface.go
Comment (ian0371) [inline governance/interface.go:110]: How about removing `CurrentHeader()`? It's not used by governance

Comment (aidan-kwon) [reply governance/interface.go:109]: Thanks! I removed `CurrentHeader` and `GetBlockByNumber`

**Review by ian0371 — APPROVED** [review]

**Review by blukat29 — APPROVED** [review]

**Comment by aidan-kwon** [comment]
@jiseongnoh Could you review this PR? I need a code owner's approval.

**Comment by jiseongnoh** [comment]
I am just curious whether it be replaced with the headerchain some day later for light client support.

**Review by jiseongnoh — APPROVED** [review]

**Comment by aidan-kwon** [comment]
> I am just curious whether it be replaced with the headerchain some day later for light client support.

I don't think so. Governance should read the current block instead of the current header.

**Pull request merged**
